internal/pkg/manifest/validate.go

```diff
@@ -734,3 +813,93 @@ func (d *DeadLetterQueue) Validate() error {
 func (*OverrideRule) Validate() error {
 	return nil
 }
+
+type validateDependenciesOpts struct {
+	mainContainerName string
+	sidecarConfig     map[string]*SidecarConfig
+	imageConfig       Image
+}
+
+type containerDependency struct {
+	dependsOn   DependsOn
+	isEssential bool
+}
+
+func validateContainerDeps(opts validateDependenciesOpts) error {
+	containerDependencies := make(map[string]containerDependency)
+	containerDependencies[opts.mainContainerName] = containerDependency{
+		dependsOn:   opts.imageConfig.DependsOn,
+		isEssential: true,
+	}
+	for name, config := range opts.sidecarConfig {
+		containerDependencies[name] = containerDependency{
+			dependsOn:   config.DependsOn,
+			isEssential: config.Essential == nil || aws.BoolValue(config.Essential),
+		}
+	}
+	if err := validateDependsOnEssentialStatus(containerDependencies); err != nil {
+		return err
+	}
+	return validateNoCircularDependencies(opts)
+}
+
+func validateDependsOnEssentialStatus(deps map[string]containerDependency) error {
+	for name, containerDep := range deps {
+		for dep, status := range containerDep.dependsOn {
+			if !deps[dep].isEssential {
+				continue
+			}
+			if err := isValidEssentialStatus(dep, strings.ToUpper(status)); err != nil {
+				return fmt.Errorf("validate %s container dependencies status: %w", name, err)
+			}
+		}
+	}
+	return nil
+}
+
+func isValidEssentialStatus(name, status string) error {
```

Comment:
Can we make these function names a bit more specific, it's hard to tell what they are doing 🥺 
```go
func validateEssentialContainerDependency(containerName, condition string) error 
```
I know that these are already existing fn names, but it's a good opportunity to clean them up